database/database_functions.py

```diff
@@ -1,64 +1,96 @@
+        '''
+        # calling the initialize function to use the specified the database
+        self.initialize(filename)
+        persons = self.SESSIONS.query(Persons).all()
+        person_dictionary = {}
+        for person in persons:
+            person_dictionary.update({person.person_name:person.person_type})
+        return person_dictionary
 
-    return al_dct
+    def get_rooms(self, filename):
+        '''
+        The function get the rooms from database and add the rooms to the dictionary with regards
+        to room name and room type to a dictionary
+        '''
+        # calling the initialize function to use the specified the database
+        self.initialize(filename)
+        rooms = self.SESSIONS.query(Rooms).all()
+        room_dictionary = {}
+        for room in rooms:
+            room_dictionary.update({room.room_name:room.room_type})
+        return room_dictionary
 
+    def get_allocations(self, filename):
+        '''
+        The function get the allocations from database and adds the allocations to the dictionary
+        with regards to person name and person type and the room they belong to into a dictionary.
+        '''
+        # calling the initialize function to use the specified the database
+        self.initialize(filename)
+        allocations = self.SESSIONS.query(Persons).all()
+        allocations_dictionary = {}
+        for allocation in allocations:
+            allocations_dictionary.update({allocation.person_name:[allocation.office_name,
+                                                                   allocation.livingspace_name]})
+        return allocations_dictionary
 
-def add_person(person_name, person_type):
-    ''' Adding a person to database'''
-    newperson = Persons(p_name=person_name, p_type=person_type)
-    SESSIONS.add(newperson)
-    SESSIONS.commit()
+    def add_person(self, filename, person_name, person_type, office_name, livingspace_name):
+        '''
+        The function is used to add the person to the dictionary
+        '''
+        # calling the initialize function to use the specified the database
+        self.initialize(filename)
+        newperson = Persons(person_name=person_name, person_type=person_type,
+                            office_name=office_name, livingspace_name=livingspace_name)
+        self.SESSIONS.add(newperson)
+        self.SESSIONS.commit()
 
-def add_room(room_name, room_type, room_capacity):
-    ''' Adding a room to database'''
-    newuser = Rooms(r_name=room_name, r_type=room_type, r_capacity=room_capacity)
-    SESSIONS.add(newuser)
-    SESSIONS.commit()
+    def add_room(self, filename, room_name, room_type):
+        '''
+        The function is used to add the rooms to the dictionary
+        '''
+        # calling the initialize function to use the specified the database
+        self.initialize(filename)
+        newuser = Rooms(room_name=room_name, room_type=room_type)
```

Comment:
Using variable name `newuser` is inappropriate, because you are adding a new room, not a user